### PR TITLE
Test key handover on SGX hardware

### PR DESCRIPTION
**Pull request by shelvenzhou**
This PR follows #810 

## Changelog

1. Add SGX local attestation in handover process to ensure both pRuntimes are on the same machine;
    - Append `sgx_target_info` to `HandoverChallenge`
    - The client needs to locally attest the info, and append the LA report in the `HandoverChallengeResponse`
    - The server can know whether they are on the same machine by verifying the LA report, and only handover the key to the local pRuntime
    - No need to ask the server to provide the LA report again in the `HandoverWorkerKey`
3. Add access control on cluster creation and forbid deploying cluster to gatekeeper;
4. Test handover in SGX hardware mode, the tested scenarios include
    - Normal cases
        - Dev mode pRuntime -> dev mode pRuntime: already covered in e2e test
        - Release pRuntime -> release pRuntime
        - Release pRuntime -> release pRuntime, and the receiver does not trust the received key until it syncs to the previous worker registration transaction
    - Abnormal cases
        - Without on-chain pRuntime binary registration: rejected
        - Handover to old version pRuntime: rejected

### CODE REVIEW

File: crates/phactory/src/prpc_service.rs [hunk 1]
```diff
@@ -1057,8 +1059,9 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
         let my_identity_key = system.identity_key.clone();
 
         // 1. verify RA report
+        // this also ensure the message integrity
         let challenge_handler = request.decode_challenge_handler().map_err(from_display)?;
-        let now_ms = system.now_ms;
+        let block_ms = system.now_ms / 1000;
```
Comment (h4x3rotab) [inline crates/phactory/src/prpc_service.rs:1064]: Why do we divide ms by 1000 here?

**Review by h4x3rotab — COMMENTED** [review]
I noticed that we attached the local attestation report to `get_worker_key_challenge` and `handover_accept_challenge`, and verify it in `handover_start`. One of the report was not verified, right?

File: crates/phactory/src/prpc_service.rs (hunk 1 above)
Comment (shelvenzhou) [reply crates/phactory/src/prpc_service.rs:1064]: Naming problem, will rename this to `block_sec`

**Comment by shelvenzhou** [comment]
> I noticed that we attached the local attestation report to `get_worker_key_challenge` and `handover_accept_challenge`, and verify it in `handover_start`. One of the report was not verified, right?

I update the changelog to explain the process. Also it can be helpful to check the example LA code https://github.com/Phala-Network/phala-blockchain/blob/master/crates/sgx-api-lite/src/lib.rs#L1-L21

**Review by h4x3rotab — COMMENTED** [review]
LGTM. Will approve when the cluster on GK is enabled for local tests

**Pull request merged**